community/io/src/test/java/org/neo4j/adversaries/pagecache/AdversarialReadPageCursor.java

```diff
@@ -232,16 +342,18 @@ public void rewind()
     @Override
     public boolean next() throws IOException
     {
-        currentReadIsInconsistent = adversary.injectFailureOrMischief( FileNotFoundException.class, IOException.class,
+        boolean currentReadIsPreparingInconsistent = state.injectFailureOrMischief( FileNotFoundException.class, IOException.class,
```

Comment:
state.injectFailureOrMischief and state.reset always come together maybe it makes sense to introduce method in state that will do both things? without external state touch